### ISSUE TITLE
CachingDirectivesExamplesSpec.createCache

### ISSUE THREAD

**Issue by raboof** [akka/akka-http#1541]
```
org.scalatest.exceptions.TestFailedException: "...mple.com/1 @ count 1[]" did not equal "...mple.com/1 @ count 1[00]"
	at org.scalatest.MatchersHelper$.indicateFailure(MatchersHelper.scala:340)
	at org.scalatest.Matchers$AnyShouldWrapper.shouldEqual(Matchers.scala:6742)
	at docs.http.scaladsl.server.directives.CachingDirectivesExamplesSpec$$anonfun$14$$anonfun$apply$18.apply(CachingDirectivesExamplesSpec.scala:141)
	at docs.http.scaladsl.server.directives.CachingDirectivesExamplesSpec$$anonfun$14$$anonfun$apply$18.apply(CachingDirectivesExamplesSpec.scala:141)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:58)
	at akka.http.scaladsl.testkit.RouteTest$$anonfun$check$1.apply(RouteTest.scala:57)
	at akka.http.scaladsl.testkit.RouteTest$$anonfun$check$1.apply(RouteTest.scala:57)
	at akka.http.scaladsl.testkit.RouteTestResultComponent$RouteTestResult.$tilde$greater(RouteTestResultComponent.scala:50)
	at docs.http.scaladsl.server.directives.CachingDirectivesExamplesSpec$$anonfun$14.apply(CachingDirectivesExamplesSpec.scala:140)
	at docs.http.scaladsl.server.directives.CachingDirectivesExamplesSpec$$anonfun$14.apply(CachingDirectivesExamplesSpec.scala:100)
```
https://jenkins.akka.io:8498/job/pr-validator-akka-http/1729/testReport/junit/docs.http.scaladsl.server.directives/CachingDirectivesExamplesSpec/createCache/

**Comment by raboof** [comment]
Is this the same problem except on Java?
```
java.lang.AssertionError: Unexpected entity! expected:<...mple.com/1 @ count 1[00]> but was:<...mple.com/1 @ count 1[]>
  Request was:      HttpRequest(HttpMethod(GET),/1,List(),HttpEntity.Strict(none/none,ByteString()),HttpProtocol(HTTP/1.1))
  Route result was: FulfilledFuture(Complete(HttpResponse(200 OK,List(),HttpEntity.Strict(text/plain; charset=UTF-8,Request for http://example.com/1 @ count 1),HttpProtocol(HTTP/1.1))))

	at akka.http.javadsl.testkit.JUnitRouteTestBase$$anon$1.reportDetails(JUnitRouteTest.scala:45)
	at akka.http.javadsl.testkit.JUnitRouteTestBase$$anon$1.assertEquals(JUnitRouteTest.scala:30)
	at akka.http.javadsl.testkit.TestRouteResult.assertEqualsKind(TestRouteResult.scala:227)
	at akka.http.javadsl.testkit.TestRouteResult.assertEntity(TestRouteResult.scala:166)
	at docs.http.javadsl.server.directives.CachingDirectivesExamplesTest.testCreateCache(CachingDirectivesExamplesTest.java:176)
```
https://jenkins.akka.io:8498/job/pr-validator-akka-http/1730/testReport/junit/docs.http.javadsl.server.directives/CachingDirectivesExamplesTest/testCreateCache/

**Comment by jonas** [comment]
There are two hard problems in computer science ... 😭

**Comment by jonas** [comment]
Yes, the Java assertion error is the equivalent of the Scala assertion in the initial comment. I wrote both of the failing tests as documentation for how to programatically configure a cache.

The tests are supposed to demonstrate that if we set a small max capacity (say 50 entries) and make twice the amount of unique requests and redo the initial request, a new cache entry should be generated.

The problem is that the cache AFAIK does not guarantee when eviction happens. From my understanding it might happen at next write time or as we clearly see at some later convenience.
